pkg/proxy/proxy.go

```diff
@@ -199,8 +186,26 @@ func (p *Proxy) RoundTrip(req *http.Request) (*http.Response, error) {
 	// Set up impersonation request.
 	rt := transport.NewImpersonatingRoundTripper(*conf, p.clientTransport)
 
+	req, remoteAddr := context.RemoteAddr(req)
+	serverDuration := time.Now()
+	clientDuration := context.ClientRequestTimestamp(req)
```

Comment:
nit: you might want to consider adopting the `k8s.io/utils` 'clock' package sometime soon - this will be tough to test otherwise 😄